integreat_cms/cms/migrations/0098_external_calendar.py

```diff
@@ -0,0 +1,119 @@
+# Generated by Django 4.2.13 on 2024-06-09 21:21
+
+import django.db.models.deletion
+from django.apps.registry import Apps
+from django.core.management.sql import emit_post_migrate_signal
+from django.db import migrations, models
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+
+from integreat_cms.cms.constants import roles
+
+
+# pylint: disable=unused-argument
+def update_roles(apps: Apps, schema_editor: BaseDatabaseSchemaEditor) -> None:
+    """
+    Add permissions for managing external calendars
+
+    :param apps: The configuration of installed applications
+    :param schema_editor: The database abstraction layer that creates actual SQL code
+    """
+    Group = apps.get_model("auth", "Group")
+    Permission = apps.get_model("auth", "Permission")
+
+    # Emit post-migrate signal to make sure the Permission objects are created before they can be assigned
+    emit_post_migrate_signal(2, False, "default")
+
+    # Clear and update permissions according to new constants
+    for role_name in dict(roles.CHOICES):
+        group, _ = Group.objects.get_or_create(name=role_name)
+        # Clear permissions
+        group.permissions.clear()
+        # Set permissions
+        group.permissions.add(
+            *Permission.objects.filter(codename__in=roles.PERMISSIONS[role_name])
+        )
```

Comment:
This code is copied from a few older migrations, which use this pattern to update the migrations. It might be worth to extract that into a common function though.

Reply:
I have added #2996 to address this now.